test/functional/terminal/highlight_spec.lua

```diff
@@ -206,4 +206,53 @@ describe('synIDattr()', function()
     eq('252', eval('synIDattr(hlID("Normal"), "fg")'))
     eq('79', eval('synIDattr(hlID("Keyword"), "fg")'))
   end)
+
+  it('returns "Normal" values for group using special fg/bg names', function()
+    execute('highlight Normal ctermfg=145 ctermbg=16 guifg=#ff0000 guibg=Black')
+    execute('highlight Visual ctermfg=bg ctermbg=fg guifg=bg guibg=fg guisp=bg')
+    eq(eval('synIDattr(hlID("Normal"), "bg")'),
+      eval('synIDattr(hlID("Visual"), "fg")'))
+    eq(eval('synIDattr(hlID("Normal"), "bg#")'),
+      eval('synIDattr(hlID("Visual"), "fg#")'))
+    eq(eval('synIDattr(hlID("Normal"), "fg")'),
+      eval('synIDattr(hlID("Visual"), "bg")'))
+    eq(eval('synIDattr(hlID("Normal"), "fg#")'),
+      eval('synIDattr(hlID("Visual"), "bg#")'))
+    eq('bg', eval('synIDattr(hlID("Visual"), "fg", "gui")'))
+    eq('bg', eval('synIDattr(hlID("Visual"), "fg#", "gui")'))
+    eq('fg', eval('synIDattr(hlID("Visual"), "bg", "gui")'))
+    eq('fg', eval('synIDattr(hlID("Visual"), "bg#", "gui")'))
+    eq('bg', eval('synIDattr(hlID("Visual"), "sp", "gui")'))
+    eq('bg', eval('synIDattr(hlID("Visual"), "sp#", "gui")'))
+  end)
+
+  it('returns "Normal" values for group using special fg/bg names \z
+    if RGB-capable UI is attached', function()
```

Comment:
a long line would be preferable to the `\z` hack :) Or try to shorten the description:

    it('fg/bg special colors resolve to "Normal" values in RGB-capable UI',